### ISSUE TITLE
Possible bug observed line 634 CActiveFinder

### ISSUE THREAD

**Issue by andrewblake1** [yiisoft/yii#2588]
This bug lies ultimately in yii's retrieval of relations from the database - findConstraints(). The solution I can see may be difficult due to potentially affecting any methods referencing the foreignKeys array as described below - however may not be too major as seems to only be referenced in 11 files including within the findConstraints method for each database implementation.

refer 634 CActiveFinder for example that was highlighting  the problem for me.

If looking for has many relations of model M within model P (pivot)
but model P contains a composite foreign key referencing table R (e.g. composite key look-up to enforce referential integrity in this circumstance)
then may get // FK defined and $pk will wrongly get set to the referenced field of model R instead of the primary key of model M which is the aim. This can then cause a fault on line 641 as the pk likely does not exist in model M.

My temporary quick fix in order to continue work was to alter line 634 to:

``` php
    if(isset($this->_table->foreignKeys[$fk]) && $this->_table->foreignKeys[$fk][0] == $parent->_table->name)  // FK defined
```
- though this would not be a good solution - just allows me to progress with my project.

however the underlying problem really is with the generation (and use) of the foreignKeys array.

Because the foreign keys array is indexed by foreign key and containing a single array of two elements that contain referenced table and referenced column, then only 1 relation can be held.

The foreign key array should be indexed like foreignKeys[$fk][$referencedTable]

This would mean the solution above would become

``` php
if(isset($this->_table->foreignKeys[$fk][$parent->_table->name]))  // FK defined
   $pk = this->_table->foreignKeys[$fk][$parent->_table->name];
```

**In Summary:**
The current implementation does not allow for multiple foreign keys on a single column - as used for enforcing referential integrity if nothing else.

If necessary - if the above hard to understand - then happy to post sql or set up a live demo of bug.

Am happy to follow the guidelines to branch and attempt fix myself if the above bug and suggested solution is accepted
